libass/ass_bitmap_engine.c

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2015 Vabishchevich Nikolay <vabnick@gmail.com>
+ *
+ * This file is part of libass.
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#include <stdlib.h>
+
+#include "ass_bitmap_engine.h"
+
+#include "config.h"
+
+#if CONFIG_ASM
+#if (defined(__i386__) || defined(__x86_64__))
+#include "x86/init.h"
+#endif
+#endif
+
+#define ENGINE_SUFFIX   c
+#include "ass_func_template.h"
+#undef ENGINE_SUFFIX
+
+void ass_bitmap_engine_init(BitmapEngine* engine, ASS_CPUFlags mask)
```

Comment:
I'm not a fan of this triple copy-paste spanning two different files. Previous approach is much cleaner and less error-prone. Indeed, setting functions without complete assembly is slightly cumbersome, but I think we should aim for the full assembly instead.